src/java.base/share/classes/java/util/concurrent/atomic/AtomicIntegerFieldUpdater.java

```diff
@@ -414,7 +414,7 @@ private static final class AtomicIntegerFieldUpdaterImpl<T>
                            !isSamePackage(tclass, caller))
                           ? caller : tclass;
             this.tclass = tclass;
-            this.offset = U.objectFieldOffset(field);
+            this.offset = U.objectFieldOffset(field); // throws IAE for static
```

Comment:
AtomicIntegerFieldUpdaterImpl already checks the field type and it is volatile. I think it would be better to expand these checks so that static volatile fields are rejected there rather than "accidentally" calling objectFieldOffset with a static field.